main/externalData/balances/index.ts

```diff
@@ -200,12 +210,22 @@ export default function (store: Store) {
     store.accountTokensUpdated(address)
   }
 
+  function handleTokenBlacklistUpdate (tokens: Token[]) {
+    const tokensToRemoveSet = new Set(tokens.map(({address, chainId}) => `${chainId}:${address.toLowerCase()}`))
+    const allBalances = storeApi.getBalances()
+    const allKnownTokens = storeApi.getKnownTokens()
+    const balancesContainsBlacklistedTokens =  allBalances.some(({address, chainId}) => tokensToRemoveSet.has(`${chainId}:${address.toLowerCase()}`))
```

Comment:
all of this logic (whether there are existing balances to remove, etc) can be moved into the individual actions. this will make them easier to test in isolation and simplify this code as well. in order to prevent unnecessary renders, you can check if any changes need to be made outside the updater function, then only update when needed.

something like this:

```
function removeKnownTokens (u: UpdaterFunction, tokens: Set<string>) {
   // check if any tokens need to be removed

   // if tokens need to be removed ... 
   u(...args, {
    ...
   })
}
```